### DISCUSSION TITLE
Setting up of relationships

### DISCUSSION THREAD

**Discussion by phifogg** [Budibase/budibase#4666] [Help]
Hi y'all,

I am new to budibase - obviously. So I am playing around with the features with a given application in mind which I would want to build during summer. So far I do like the interface and it seems to be very flexible.

Currently, I have a hard time getting my data model in order. Here is what I want:
Table 1: participiants
Table 2: events
Table 3: registrations

So I want a list of events and potential participiants and events. Then I would love to have a registration form where a participiant could register himself for a given event. To model that I would think the registration table would need to columns with foreign keys to participiants and events.

How would I create that from budibase?

When I use the 'create column' feature with type relationship, it does create some 'fk_...' columns on both sides. But creating data i cannot really select something.

When I use the 'define existing relationship' I seem to get a better result. Still it is creating columns on both sides of the relationship. I.e. on my participiants table I suddenly get an registration column which I do not need.

I am sure there is some easy piece I am missing.... not?

**Comment by mjashanks** [comment]
I think I see what you mean - I set up the same model as you laid out above. In my "Edit Row" screen, I see..

![image](https://user-images.githubusercontent.com/3524181/155501017-eff8182f-3a3c-4197-92ec-0061e3449ff4.png)

.. and the same on the table headers and "New Row" screen.

I think that this is just a display issue - i.e. the `fk_...` fields should not appear. I will log this as a bug.

When I select from the `Participiant` and `Event` dropdowns, the `fk_` field get set to the correct `id` values.

So, just to explain what all the fields are
- `fk_participiant_registrations` and `fk_event_registrations` are foreign key columns that have been created in the SQL database. These should hidden, and controlled by the `Participiant` and `Event` fields
- `Participiant` and `Event` fields are **NOT** columns in the SQL database. They are really just alias / display columns, akin to the following....

```sql
select 
   Registrations.*,
   Participiant.name as Participiant
from Registrations
inner join Participiant on Registrations.fk_participiant_registrations = Participiant.id
```
i.e. `Participiant.name as Participiant` is that alias that I refer to above.

**Reply by phifogg** [reply]
Thanks for your answer. I do understand the fk-fields :D 
Looking forward for the bug to be fixed.
